login/login-tool/tool/src/webapp/WEB-INF/xlogin-context.saml.adfs-prod.xml

```diff
@@ -216,7 +216,11 @@
     </bean>
 
     <!-- SAML 2.0 WebSSO Assertion Consumer -->
-    <bean id="webSSOprofileConsumer" class="org.springframework.security.saml.websso.WebSSOProfileConsumerImpl"/>
+    <bean id="webSSOprofileConsumer" class="org.springframework.security.saml.websso.WebSSOProfileConsumerImpl">
+        <!-- Duration in seconds after initial authentication with the IDP that SSO login will be allowed: see
+             https://docs.spring.io/spring-security-saml/docs/current/reference/html/configuration-advanced.html#time-interval -->
+        <property name="maxAuthenticationAge" value="7200"/>
```

Comment:
In my experience, ADFS by default can have very old authenticationAge. Should we set this to a full day on the adfs file only?

Reply:
We could do that. Locally we set this to 86400. I don't have any experience with other SAML IdPs - what sort of defaults do other systems have?

It did seem like if the IdP thought your authentication was too old, it would prompt you to re-authenticate before redirect back to Sakai though? So doesn't seem any harm in setting this to a large value unless you want to enforce stricter timeouts than the IdP which leads to a somewhat poor user experience because they get prompted to login via Sakai (xlogin) when they expect to already be authenticated.

Reply:
I've never seen an old value coming from Shibboleth. My instinct is that Shibboleth sets the authentication timestamp to the current time when a user has an existing session while ADFS sets the authentication timestamp to the *original* time of login. So yeah, let's set this to 86400 for ADFS and it should help ADFS schools out.

Reply:
OK, changed to 86400 for ADFS.